app/models/ems_refresh/save_inventory_infra.rb

```diff
@@ -332,9 +332,10 @@ def save_switches_inventory(host, hashes)
   end
 
   def remove_obsolete_switches
-    # delete from switches as s where s.shared is NULL and s.id not in (select switch_id from host_switches)
-    # delete from switches as s where s.shared = 't' and s.id not in (select switch_id from host_switches)
-    Switch.where.not(:id => HostSwitch.all.collect(&:switch).uniq).destroy_all
+    # delete from switches as s where s.id not in (select switch_id from host_switches)
+    # AND s.type not like '%PhysicalInfraManager%'
+    switch_ids = HostSwitch.all.collect(&:switch_id)
+    Switch.where.not(:id => switch_ids).where.not("type LIKE '%PhysicalInfraManager%'").destroy_all
```

Comment:
Could you do just `Switch.where.not(:id => switch_ids, :type => 'PhysicalSwitch').destroy_all`

Reply:
hey @agrare type is a little bigger `ManageIQ::Providers::Lenovo::PhysicalInfraManager::PhysicalSwitch` and if I use this `Switch.where.not(:id => switch_ids, :type => "ManageIQ::Providers::Lenovo::PhysicalInfraManager::PhysicalSwitch").destroy_all` I will only to fix Lenovo PhysicalSwitches and not for all PhysicalInfraStructure. What do you think?

Reply:
Also, today we have a new PhysicalInfraStructure the RedFish provider.

Reply:
Okay how about this, before we save any inventory, up [here](https://github.com/CharlleDaniel/manageiq/blob/f880ad7288241343fb8e31ef078c8e05db3690e7/app/models/ems_refresh/save_inventory_infra.rb#L67) we collect all switch ids up front by doing `ems.switches.pluck(:id)` which we then pass into this method.  That will give us the scope of switches that we are going to delete here and not impact other providers at all.